pod/perlretut.pod

```diff
@@ -2853,7 +2853,7 @@ for another vowel. Thus, match or no match makes no difference, and the
 regexp engine proceeds until the entire string has been inspected.
 (It's remarkable that an alternative solution using something like
 
-   $count{lc($_)}++ for split('', "supercalifragilisticexpialidocious");
+   $count2{lc($_)}++ for split('', "supercalifragilisticexpialidocious");
```

Comment:
I'd rather see `count2` replaced by `count` in the next line. Also, `lc($_)` should be changed to `$_` to make the two snippets truly equivalent.

Reply:
The `lc` is needed AFAICT to get the effect that the `/i` in the prior snippet has.

I think the second snippet should begin with the line `%count = ();` to indicate a fresh hash, and then remove the `2` from the instance that has it.  But I would not oppose making all instances in this snippet refer to `count2`, but I agree with @mauke that without the 2 is better, but not by much.

But something should be done, as the current example is broken

Reply:
> The `lc` is needed AFAICT to get the effect that the `/i` in the prior snippet has.

No, the previous snippet is `/([aeiou])(?{ $count{$1}++; })(*FAIL)/i`, which (modulo Unicode) is the same as `/([aAeEiIoOuU])(?{ $count{$1}++; })(*FAIL)/`. The `/i` only affects what the pattern matches; `$1` is still set to whatever was originally in the target string.

(And then someone should re-benchmark the thing to see which version is actually faster, especially on longer strings.)